### DISCUSSION TITLE
Doubt about the calculated thrust in the Mellinger controller.

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#537] [Q&A] [answered]
Hi.

Sorry for the inconvenience, but I would like to know how you concluded that current_thrust has to be multiplied by massThrust =132000 in the mellinger control only in the case when thrust has to be calculated on board?.

I'd appreciate your help.

Regards.

**Comment by hmllr** [comment]
Hi :)
This factor is needed for mapping the thrust to a PWM value (the motors are controlled via PWM, and the PWM signals work with an uint16 counter). Unfortunately I do not know why it is exactly 132000 - partly this is also a design parameter, meaning if you change the weight, have other motors, or other changes you would need to tune this again, so I think this is partly experimental evaluated - but maybe @whoenig has something to add here?
Hope this helps already,
Hanna

**Comment by magalhaesw** [comment, marked as answer]
Hi ! 
I had the same question at : https://github.com/orgs/bitcraze/discussions/513

In one of the replies, 
@whoenig said:
"Yes, the high gains in Mellinger are because they are not based on SI units. They were found empirically."

**Reply by Wallat1** [reply]
Hi.

Thank you very much for your help.

Regards.
